### PR TITLE
fix: remove redundant stdout monkey-patch causing EPIPE crashes

### LINKED ISSUE #863: fix: remove redundant stdout monkey-patch causing EPIPE crashes in nextjs instrumentation
## Problem

`FrontmanNextjs__LogCapture.interceptStdout` monkey-patches `process.stdout.write` to capture build tool output matching patterns (`webpack`, `turbopack`, `Compiled`, `Failed`). This wrapper has no error handling on the forwarded write call, causing fatal EPIPE crashes when the stdout pipe closes.

**Sentry issue:** [ELIXIR-5D](https://frontman-gm.sentry.io/issues/ELIXIR-5D)

### Stack trace

```
Error: write EPIPE
  at afterWriteDispatched (node:internal/stream_base_commons:159:15)
  ...
  at @frontman-ai/nextjs/dist/instrumentation.js:2198:14 (process.stdout.write)
  at writeLine (next/dist/server/dev/log-requests.js:126:20)
```

Next.js dev server logs an HTTP request via `process.stdout.write`, our wrapper forwards the call, the pipe is closed → unhandled EPIPE → process crash.

## Root Cause

The wrapper (`FrontmanNextjs__LogCapture.res:208-217`) replaces `process.stdout.write` with a plain function that:
1. Has no error handling on the `originalWrite` call
2. Bypasses Node's internal stream error handling
3. Doesn't respect the `Writable.write` overloaded signature (uses `...args` instead of explicit params)

## Why It's Also Redundant

Research into Next.js internals (tested against `next@^15.5.14`) confirms:

- **All four patterns go through `console.*`**, not `process.stdout.write` directly. Next.js's `Log` utility (`next/dist/build/output/log.js`) routes `Compiled`, `Failed`, webpack, and turbopack messages through `console.log/warn/error`.
- **The only direct `stdout.write` caller** is `log-requests.js` for HTTP request logs (`GET /page 200 in 15ms`) — which don't match any of our patterns.
- **Messages are captured twice** — `console.log("Compiled...")` hits our console intercept, then internally calls `process.stdout.write`, hitting our stdout intercept. Same message, two buffer entries (`Console` vs `Build` level).

## Fix

1. **Delete `interceptStdout`** from `FrontmanNextjs__LogCapture.res` — zero data loss, eliminates crash, removes duplication
2. **Optionally**: add pattern matching inside the existing console intercept to tag matching messages with `Build` level, preserving the semantic distinction
3. **Add tests** for the failure mode — the test file has zero coverage for stdout interception or write failures
4. **Audit `FrontmanCore__LogCapture.res`** — same pattern exists there wrapping both stdout and stderr (stderr capture is needed for Astro `[ERROR]` messages, so that one needs the EPIPE guard rather than removal)

## Files

- `libs/frontman-nextjs/src/FrontmanNextjs__LogCapture.res` — lines 208-217 (delete)
- `libs/frontman-nextjs/test/FrontmanNextjs__LogCapture.test.res` — add missing coverage
- `libs/frontman-core/src/FrontmanCore__LogCapture.res` — audit stderr wrapper for same vulnerability

### PR DESCRIPTION
**Pull request by BlueHotDog**
## Summary

Closes #863

- **Remove `interceptStdout`** from NextJS LogCapture — it monkey-patched `process.stdout.write` with no error handling, causing fatal EPIPE crashes when the pipe closes. It was also redundant: all four captured patterns (`webpack`, `turbopack`, `Compiled`, `Failed`) go through `console.*` in Next.js, not `process.stdout.write` directly
- **Add build-level tagging in console handlers** — messages matching `stdoutPatterns` are now tagged as `Build` level instead of `Console`, preserving the semantic distinction with zero data loss
- **Add EPIPE guard to Core LogCapture** — the Core version wraps both stdout and stderr (stderr capture is needed for Astro `[ERROR]` messages), so it gets a try/catch that swallows EPIPE and re-throws other errors
- **Add tests** — 10 new tests across both libs covering build-level tagging and EPIPE handling

## Test plan

- [x] NextJS LogCapture: 177/177 tests pass (`libs/frontman-nextjs`)
- [x] Core LogCapture: 298/298 tests pass (`libs/frontman-core`)
- [x] Pre-push reanalyze: 0 issues across all packages
- [ ] Verify Sentry ELIXIR-5D stops firing after deploy

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/frontman-ai/frontman/pull/865" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: libs/frontman-core/src/FrontmanCore__LogCapture.res [hunk 1]
```diff
@@ -140,41 +140,55 @@ let addLog = (
...
 }
 
+let detectLevel = (state: state, message: string): logLevel => {
+  let matchesBuildPattern =
+    state.config.stdoutPatterns->Array.some(pattern => message->String.includes(pattern))
+  switch matchesBuildPattern {
+  | true => Build
+  | false => Console
+  }
+}
```
[the remark this answers is not in the document]
Comment (BlueHotDog) [reply libs/frontman-core/src/FrontmanCore__LogCapture.res:152]: The `detectLevel` addition is intentional — it compensates for removing the redundant stdout intercept. Previously, build output captured via the stdout monkey-patch got labeled `Build`. Now that stdout interception is removed (nextjs) or simplified (core), the console handlers need to do that classification instead.

Regarding the broad patterns (`"error"`, `"Error"`, etc.): these match on message *content* from `argsToString`, not on the console method name. So `console.error("User not found")` produces message `"User not found"` which does **not** match. Messages that *do* contain "Error" (e.g. `"TypeError: undefined is not a function"`) are arguably build-relevant anyway.

Not a perfect heuristic, but not actively broken. We can refine the pattern lists separately if real misclassification shows up.

[the remark this answers is not in the document]
Comment (BlueHotDog) [reply libs/frontman-core/src/FrontmanCore__LogCapture.res:152]: Valid catch. Both interceptors were producing duplicate Build entries for console-originated messages flowing through stdout.

Fixed in b72723fc9: added a reentrant guard (`insideConsoleHandler` ref on state). Console handler sets it before calling `originalLog`, `handleStdoutWrite` skips when the flag is set. Direct `process.stdout.write` from build tools still captured normally.

Also added a dedup assertion test that activates both interceptors and verifies exactly 1 Build entry per console.log call.
